woocommerce/woocommerce-ios
# Read-only Refunds - Networking models

Closes #1307 

In this PR: preparing the Networking subproject to handle refunds. 
⚠️ the main project will not build
⚠️ the main project unit tests will fail

(I will not merge this until the other PRs are: able to successfully build, all unit tests pass, are reviewed, and are ready for merge.)

This PR is absurdly long. @jaclync I followed along on the old PR review for Paolo's Refunds 1 and made the changes here. LMK if I missed any! 

## Changes
- Add new models, `OrderItemTaxRefund` and `OrderItemRefund`
- Add `OrderItemTax` to the `OrderItem`
- Add Codable to `OrderItemTaxRefund` and `OrderItemRefund`
- `Refund` model conforms to `Codable` (`Decodable & Encodable`)
- Restore Paolo's RefundsRemote class
- Write unit tests for RefundMapper and RefundListMapper

### To test
- point the project to the Networking target
- build and run (it should build the subproject successfully)
- run the unit tests for Networking (they should all pass)

Update release notes:
